modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/ProjectAccessSiteProcessor.java

```diff
@@ -51,38 +49,33 @@ public ProcessUpdateResult processUpdate( final ProcessUpdateParams params )
         final SiteConfigs editedSiteConfigs = editedSite.getSiteConfigs();
         final Site originalSite = (Site) params.getOriginalContent();
         final SiteConfigs originalSiteConfigs = originalSite.getSiteConfigs();
-        final User modifier = params.getModifier();
 
-        if ( !Objects.equals( originalSiteConfigs, editedSiteConfigs ) && !this.hasContentAdminRole( modifier ) )
+        final Context context = ContextAccessor.current();
+        final AuthenticationInfo authenticationInfo = context.getAuthInfo();
+        final ProjectName projectName = ProjectName.from( context.getRepositoryId() );
+
+        if ( !Objects.equals( originalSiteConfigs, editedSiteConfigs ) )
         {
-            throw new RoleRequiredException( modifier.getKey(), RoleKeys.ADMIN, RoleKeys.CONTENT_MANAGER_ADMIN );
+            if ( !this.projectPermissionsContextManager.hasAdminAccess( authenticationInfo ) )
+            {
+                if ( ProjectConstants.DEFAULT_PROJECT_NAME.equals( projectName ) )
+                {
+                    throw new ProjectAccessRequiredException( authenticationInfo.getUser().getKey() );
+                }
+                else if ( !this.projectPermissionsContextManager.hasAnyProjectPermission( projectName, authenticationInfo,
+                                                                                          Set.of( ProjectRole.OWNER ) ) )
+                {
+                    throw new ProjectAccessRequiredException( authenticationInfo.getUser().getKey(), ProjectRole.OWNER );
+                }
+            }
         }
 
         return null;
```

Comment:
@vbradnitski why this method always returns null?

Reply:
It's a processor, which called during create/update methods in a queue of similar classes. So it should check authInfo and throw an exception if it doesn't contain project role. Result of this particular processor is never used.